src/ppx/command_line_parser.cpp

```diff
@@ -131,14 +121,24 @@ std::optional<CommandLineParser::ParsingError> CommandLineParser::Parse(int argc
         if (res->second.find('=') != std::string_view::npos) {
             return "Unexpected number of '=' symbols in the following string: \"" + std::string(argString) + "\"";
         }
-        if (StartsWithDoubleDash(res->first) && res->first == "--" + mJsonConfigFlagName) {
-            mOpts.AddOption(mJsonConfigFlagName, ppx::string_util::TrimBothEnds(res->second));
-            continue;
-        }
         args.emplace_back(res->first);
         args.emplace_back(res->second);
     }
 
+    // Another pass to identify JSON config files, add that option, and remove it from the argument list
+    std::vector<std::string_view> argsFiltered;
+    for (size_t i = 0; i < args.size(); ++i) {
+        bool nextArgumentIsParameter = (i + 1 < args.size()) && !StartsWithDoubleDash(args[i + 1]);
```

Comment:
One thing that bugs me: it seems this is the only flag which allows `--flag value`.
Seems like all the other flags have the format `--flag=value`.
Seems a bit inconsistent.

I'm fine sending this for now, but once you work on the error handling in flags, I believe this should be sanitized.
There is one ambiguous case if you allow `--flag value`: the booleans.
For SPIRV-Tools, `--bool-flag` cannot have a second value.
so it's either `--bool-flag`, `--no-bool-flag` or `--bool-flag=false`.
But for all other flags, it can be `--flag=value` and `--flag value`